### PR TITLE
fix: Concurrent in NewTracer and GetTracer methods.

### PR DESCRIPTION
**Pull request by maxingg**
**What this PR does**:
Creating and retrieving tracers and holders has thread safety issues in concurrent cases, which are fixed using sync.Map and CAS.

### CODE REVIEW

File: pkg/server/otel.go [hunk 1]
```diff
@@ -87,9 +87,13 @@ func NewTracer(name tracing.ProtocolName) (tracing.Trace, error) {
 	}
 	holder := driver.GetHolder(name)
 
+	old := holder.ID
+	for !atomic.CompareAndSwapUint64(&holder.ID, old, old+1) {
+		old = holder.ID
```
Comment (AlexStocks) [inline pkg/server/otel.go]: 是不是可以加上 time.Sleep(50e6) 这个语句啊？别搞不好成了死循环

Comment (AlexStocks) [reply pkg/server/otel.go]: 至少别太耗费 cpu

Comment (maxingg) [reply pkg/server/otel.go]: 已经改成原子自增操作

**Comment by mark4z** [comment]
fix go format

**Review by ztelur — APPROVED** [review]

**Review by AlexStocks — APPROVED** [review]

**Review by mark4z — APPROVED** [review]

**Pull request merged**